src/Widgets/Popovers/AvatarPopover.vala

```diff
@@ -84,18 +84,17 @@ namespace SwitchboardPlugUserAccounts.Widgets {
             file_dialog.update_preview.connect (() => {
                 string uri = file_dialog.get_preview_uri ();
                 if (uri != null && uri.has_prefix ("file://")) {
-                    preview_area.set_from_gicon_async.begin (
-                        new FileIcon (file_dialog.get_file ()),
-                        256,
-                        null,
-                        (obj, res) => {
-                            try {
-                                preview_area.set_from_gicon_async.end (res);
-                                preview_area.show ();
-                            } catch (Error e) {
-                                preview_area.hide ();
-                            }
-                        });
+                    try {
+                        Gdk.Pixbuf pixbuf = new Gdk.Pixbuf.from_file_at_scale (
+                            file_dialog.get_file ().get_path (),
+                            250,
+                            250,
```

Comment:
I think this was intentionally 256 to match the pixel size of the preview area

```suggestion
                            preview_area.pixel_size,
                            preview_area.pixel_size,
```

I think with Pixbuf we have to handle display scaling ourselves as well so we might have to multiply this by the display scale for HiDPI

Reply:
Yes, that makes sense. Let me accept this suggestion and I will test it on HiDPI for weirdness and follow up!